### PR TITLE
Args handling using properties EVA-357

### PR DESCRIPTION
**Pull request by poggio84**
More Spring-compliant pipeline arguments handling:
- use of property file to hold job options
- properties are loaded using spring
- properties are shared across different steps using autowired objects
- common configuration property used in tests
- there are 2 different type of args: pipeline (used in pipeline only) and variant (used by opencga)
- listeners removed as no longer necessary

https://www.ebi.ac.uk/ena/jira/browse/EVA-357

### CODE REVIEW

File: eva-pipeline/src/main/java/embl/ebi/variation/eva/VariantJobsArgs.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+ * @author Diego Poggioli &lt;diego@ebi.ac.uk&gt;
+ *
+ * TODO: 20/05/2016 add type validator?
+ */
+@Component
+public class VariantJobsArgs {
+    private static final Logger logger = LoggerFactory.getLogger(VariantJobsArgs.class);
+
+    ////common
+    @Value("${input:}") private String input;
```
Comment (cyenyxe) [inline eva-pipeline/src/main/java/embl/ebi/variation/eva/VariantJobsArgs.java:43]: Will this provide a default "", preventing the pipeline to break if no input is provided? It should break.

**Comment by cyenyxe** [comment]
Be careful with the copyright notice, it should say 2016 instead of 2015.

File: eva-pipeline/src/main/java/embl/ebi/variation/eva/VariantJobsArgs.java (hunk 1 above)
Comment (poggio84) [inline eva-pipeline/src/main/java/embl/ebi/variation/eva/VariantJobsArgs.java:43]: It brakes later on in opencga... We should add a layer of args/properties validation to check the null/empty, format and input file/dirs

File: eva-pipeline/src/test/java/embl/ebi/variation/eva/pipeline/jobs/VariantConfigurationTest.java [hunk 2]
```diff
@@ -133,66 +133,51 @@ public void validTransform() throws JobExecutionException, IOException {
...
+        JobExecution execution = jobLauncher.run(job, getJobParameters());
+
+        assertEquals(input, pipelineOptions.getString("input"));
         assertEquals(ExitStatus.FAILED.getExitCode(), execution.getExitStatus().getExitCode());
     }
 
-    @Test
+/*    @Test
     public void validLoad() throws JobExecutionException, IllegalAccessException, ClassNotFoundException,
             InstantiationException, IOException, StorageManagerException {
```
Comment (cyenyxe) [inline eva-pipeline/src/test/java/embl/ebi/variation/eva/pipeline/jobs/VariantConfigurationTest.java:159]: Does this test (and the other validLoadXXX) run if uncommented? If they have been moved to `VariantLoadConfigurationTest` then please remove these commented blocks.

Comment (poggio84) [inline eva-pipeline/src/test/java/embl/ebi/variation/eva/pipeline/jobs/VariantConfigurationTest.java:159]: I've removed the comment and leave the tests to run.. I also added a todo comment and I will review this in https://www.ebi.ac.uk/ena/jira/browse/EVA-387

File: eva-pipeline/src/main/resources/application.properties [hunk 3]
```diff
@@ -1 +1,55 @@
...
+annotate=false
+includeStats=false
+
+#SKIP STEPs
+#skipLoad
+#skipStatsCreate
+#skipStatsLoad
+skipAnnotGenerateInput=true
+skipAnnotCreate=true
+skipAnnotLoad=true
```
Comment (cyenyxe) [inline eva-pipeline/src/main/resources/application.properties:55]: This `skipAnnotXXX" properties set to true overwrite the defaults in`VariantJobsArgs.java`. Do you want me to write the most sensible defaults for this?

Comment (poggio84) [inline eva-pipeline/src/main/resources/application.properties:55]: Yes the most common scenario should be in the property file... this one is just random to be able to test the process

**Pull request merged**